apps/examples/testcase/ta_tc/media/itc/itc_media_bufferoutputdatasource.cpp

```diff
@@ -0,0 +1,139 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <media/BufferOutputDataSource.h>
+#include "tc_common.h"
+#include <errno.h>
+
+using namespace std;
+using namespace media;
+using namespace media::stream;
+
+static unsigned int channels = 2;
+static unsigned int sampleRate = 16000;
+static audio_format_type_t pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+
+static void itc_media_BufferOutputDataSource_getChannels_setChannels_p(void)
+{
+	unsigned int compare_channels = 3;
+	BufferOutputDataSource dataSource(channels, sampleRate, pcmFormat);
+	TC_ASSERT_EQ("getChannels", dataSource.getChannels(), 2);
+	dataSource.setChannels(compare_channels);
+	TC_ASSERT_EQ("getChannels", dataSource.getChannels(), compare_channels);
+	dataSource.setChannels(0);
+	TC_ASSERT_EQ("getChannels", dataSource.getChannels(), 0);
+	TC_SUCCESS_RESULT();
+}
+
+static void itc_media_BufferOutputDataSource_getSampleRate_setSampleRate_p(void)
+{
+	audio_sample_rate_t sample_rate[] = { AUDIO_SAMPLE_RATE_7350,
+		AUDIO_SAMPLE_RATE_8000,
+		AUDIO_SAMPLE_RATE_11025,
+		AUDIO_SAMPLE_RATE_12000,
+		AUDIO_SAMPLE_RATE_16000,
+		AUDIO_SAMPLE_RATE_22050,
+		AUDIO_SAMPLE_RATE_24000,
+		AUDIO_SAMPLE_RATE_32000,
+		AUDIO_SAMPLE_RATE_44100,
+		AUDIO_SAMPLE_RATE_48000,
+		AUDIO_SAMPLE_RATE_64000,
+		AUDIO_SAMPLE_RATE_88200,
+		AUDIO_SAMPLE_RATE_96000
+	};
+	int enum_size = sizeof(sample_rate) / sizeof(sample_rate[0]);
+	int enum_counter = 0;
+
+	BufferOutputDataSource dataSource(channels, sampleRate, pcmFormat);
+	TC_ASSERT_EQ("getSampleRate", dataSource.getSampleRate(), sampleRate);
+
+	for (enum_counter=0; enum_counter<enum_size; enum_counter++) {
```

Comment:
Spaces between operator and operands still needs to be taken care of!

Reply:
changes done and code also updated in gerrit